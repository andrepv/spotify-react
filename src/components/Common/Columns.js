import React, { Component } from "react";
import PropTypes from "prop-types";

export default class Columns extends Component {
  render() {
    const {amount = 1, children} = this.props;
    const columnCount = amount;
    if (columnCount === 1) {
     return children;
    }

    const itemCount = React.Children.count(children);
    const columnItems = Math.ceil(itemCount / columnCount);
    const items = React.Children.toArray(children);

    return [...Array(columnCount).keys()].map((column, index) => {
      return (
        <div className="column" key={index}>
          {items.slice(column * columnItems, columnItems * (column + 1))}
        </div>
      );
    });
  }
}

Columns.propTypes = {
  amount: PropTypes.number.isRequired,
};