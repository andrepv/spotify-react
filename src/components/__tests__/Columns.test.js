import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import Columns from "components/Common/Columns";

describe("<Columns />", () => {
  it("renders correctly", () => {
    const component = renderer.create(
      <Columns amount={2}>
        <div>Block 1</div>
        <div>Block 2</div>
        <div>Block 3</div>
        <div>Block 4</div>
      </Columns>
    ).toJSON();
    expect(component).toMatchSnapshot();
  });

  it("should not render columns", () => {
    const wrapper = mount(
      <Columns amount={1}>
        <div>Block 1</div>
        <div>Block 2</div>
        <div>Block 3</div>
      </Columns>
    );
    expect(wrapper.find(".column")).toHaveLength(0);
  });

  it("should wrap every block in the single column", () => {
    const wrapper = mount(
      <Columns amount={3}>
        <div>Block 1</div>
        <div>Block 2</div>
        <div>Block 3</div>
      </Columns>
    );
    expect(wrapper.find(".column")).toHaveLength(3);
    expect(wrapper.find(".column").everyWhere(n => {
      return n.children().length === 1;
    })).toEqual(true);
  });

  it("the first column should have the two blocks", () => {
    const wrapper = mount(
      <Columns amount={2}>
        <div>Block 1</div>
        <div>Block 2</div>
        <div>Block 3</div>
      </Columns>
    );
    expect(wrapper.find(".column")).toHaveLength(2);
    expect(wrapper.find(".column:first-child").children()).toHaveLength(2);
    expect(wrapper.find(".column:last-child").children()).toHaveLength(1);
  });
});