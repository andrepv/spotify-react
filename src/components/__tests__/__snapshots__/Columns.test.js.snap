// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Columns /> renders correctly 1`] = `
Array [
  <div
    className="column"
  >
    <div>
      Block 1
    </div>
    <div>
      Block 2
    </div>
  </div>,
  <div
    className="column"
  >
    <div>
      Block 3
    </div>
    <div>
      Block 4
    </div>
  </div>,
]
`;
